internal/controller/datadogagent_controller.go

```diff
@@ -68,6 +68,11 @@ type DatadogAgentReconciler struct {
 // +kubebuilder:rbac:groups=authorization.k8s.io,resources=rolebindings,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups="",resources=pods/exec,verbs=create
 
+// Finalizer (cluster-scoped resources)
+// Note: we already have verbs=* for apiservices (External Metrics server feature requires it), so this is not needed.
+// However, we keep it here for clarity.
+// +kubebuilder:rbac:groups=apiregistration.k8s.io,resources=apiservices,verbs=deletecollection
```

Comment:
As we already have `*` on `apiregistration.k8s.io/apiservices`, do we want to still keep this section ?